tools/unix/clangd-check.sh

```diff
@@ -5,10 +5,46 @@
 
 output_dir=$(mktemp -d)
 top_of_tree=$(git rev-parse --show-toplevel)
+declare -a job_pids
+
+# Guess at number of CPUs (could be probed, but portable requires work).
+jobs_max=8
+
+# Check bash version to see if `wait -n` is supported. This allows waiting on first of any
+# process specified as an argument to wait to exit rather than waiting for all the processes
+# specified to exit. (ie batch vs stream).
+bash_major=$(echo $BASH_VERSION | sed -e 's/\..*/0000/')
+bash_minor=$(echo $BASH_VERSION | sed -e 's/^[^.]*\.//' -e 's/\..*//')
+bash_linear=$(($bash_major + $bash_minor))
+if [ ${bash_linear} -ge 50001 ]; then
+  wait_args="-n"
+fi
 
 cd "${top_of_tree}"
 for i in $(find . -name '*.h' -o -name '*.c++'); do
+    if [ ${#job_pids[*]} -eq ${jobs_max} ]; then
```

Comment:
maybe `xargs -P` is a better choice here?

Reply:
Changing the body to be something close to what is desired with `xargs -P`:
```
find . -name '*.h' -o -name '*.c++' | xargs -P 8 -n 1 -t bash -c "clangd --check={} 2>/dev/null"
```
then this script takes 8m 5s to complete locally for ew, switching to /bin/sh instead of bash takes 7m 59s. As-is, the script takes 7m 8s for ew.

The difference is probably down to the added cost of spawning a few thousand shell instances. The quick test here uses a shell as we're going to want to pipe the output somewhere and I'm not sure there's a good way to do that on the `xargs -P` path without a shell. Happy to be taught to fish, I rarely never use `xargs -P`.